### ISSUE TITLE
macos install help

### ISSUE THREAD

**Issue by molleweide** [david-janssen/kmonad#220]
Hi,

If I already have Karabiner should I exclude the `--extra-include-dirs=c_src/mac/Karabiner-VirtualHIDDevi...` flag when running `stack`?

Also, do I have to turn off Karabiner in order to use kmonad?
I'm super excited to try this!!

Thanks!

**Comment by thoelze1** [comment]
Yes and yes! Good luck I hope you like it

**Comment by molleweide** [comment]
Thanks for replying!
Can I reinstall it by doing it again. You see, before I wrote the issue i ran the command with the fag :P

**Comment by thoelze1** [comment]
My mistake, I read "exclude" instead of "include." It sounds like you did it correctly. If there were no errors you should be good to go!

**Comment by molleweide** [comment]
allright cool!
now I can see that it doesn't seem to be in path after install.
can I see in the stack yaml or cabal file where the build is it located? Do you know this?

**Comment by molleweide** [comment]
now it works!!!!!!!!!!!!!

**Comment by molleweide** [comment]
and karabiner also seems to be working without any problems at all. i haven't configured anything
with kmonad yet tho...

**Comment by molleweide** [comment]
allright! I will get back later when i have tried out some configs. my goal is to mirror my ergodox

**Comment by molleweide** [comment]
Hi again,

Now I have read through the docs a little bit and familiarized myself a little bit. I read through the #185 issue which was a very nice read.
My kmonad is recognized and running kmonad -/path/to/apple.kbd returns successfully it seems, but when I start kmonad this is the message that is outputted:
```
> kmonad path/to/apple.kbd
IOServiceGetMatchingService error
IOHIDDeviceOpen error: (iokit/common) privilege violation
IOHIDDeviceOpen error: (iokit/common) privilege violation
```
I will search through the issues history some more to see if i can find solution.

**Comment by molleweide** [comment]
found https://github.com/david-janssen/kmonad/issues/105

**Comment by molleweide** [comment]
are you guy using nyxt browser also? i think that it is cool that nyxt is written in lisp, kmonad in haskell, and openscad for keyboard printing is clojure. did you build your dactyl?

**Comment by molleweide** [comment]
> found #105

okay, so this is probably the reason for my error but this is quite new stuff to me so I am going to have to read through the issues a couple of times.

**Comment by molleweide** [comment]
now i have added kmonad executable to input monitoring but it is not working yet it seems.
error persists.

**Comment by molleweide** [comment]
also when I run `kextstat | grep Karabiner` it returns empty no errors fyi

**Comment by molleweide** [comment]
maybe I am having the same problem as this https://github.com/david-janssen/kmonad/issues/105#issuecomment-721898449

**Comment by molleweide** [comment]
wow you also have modifier keys on homerow. that is one of the things I want to copy from my ergodox. and also layer hold switches on zxcvb row.

**Comment by thoelze1** [comment]
> are you guy using nyxt browser also?

A year ago I tried using and building it on Mac, which had some issues at the time. I was (and still am) inexperienced in Common Lisp so I gave up at some point. I'm still in the process of migrating fully to GNU/Linux at which point I will give it another try

> did you build your dactyl?

I bought a dactyl manuform a few months ago but I don't yet use it regularly

> Do I have to run sudo before. But it seems this is only for startup right?

I think you need to use `sudo` (`sudo` is missing from the script in #105 because that script is run as root, so `sudo` would be redundant). Once the kext is loaded it will stay loaded until reboot unless it's unloaded for some other reason

**Comment by molleweide** [comment]
yeah i delved into virtualbox a couple of days ago now because I wanted to test kmonad and nyxt. most other stuff exist on mac already. i got nyxt working yesterday and it is really coolactually. feels like it has super active development. i get issues in my feed from the core devs multiple times a day. that seems like a lot to me but i am also just a beginner sort of. one app that is really aweome that doesn't work on linux atm is Vimac. If you haven't tried linux on mac w virtualbox i can tell you that it is really easy. 

ok i'm gonna test with sudo then.

**Comment by molleweide** [comment]
hmmm, after doin `sudo /sbin/kextload ...` it still throws this:
```
/Library/Application failed to load - (libkern/kext) not found; check the system/kernel logs for errors or try kextutil(8).
/Users/hjalmarjakobsson/.dotfiles/Support/org.pqrs/Karabiner-VirtualHIDDevice/Extensions/org.pqrs.driver.Karabiner.VirtualHIDDevice.v061000.kext failed to load - (libkern/kext) not found; check the system/kernel logs for errors or try kextutil(8).
```

**Comment by molleweide** [comment]
wait path in the err message seems to be wrong. it goes into my dotfies!! i'll check it out

**Comment by molleweide** [comment]
yo i solved the problem. it was a typo i believe. i am still a bit unsure but i think the error was because I wrote `.kext/` and forgot to rm the slash.

**Comment by molleweide** [comment]
dude i am testing out your mbp.kbd now!!!

**Comment by molleweide** [comment]
For how long have you been using kmonad?
